molgenis-data-mysql/src/test/java/org/molgenis/data/mysql/MysqlRepositoryTest.java

```diff
@@ -115,6 +117,36 @@ public void addStreamUpdateStreamFindAll()
 	}
 
 	@Test
+	public void testDeleteAllForXrefs()
+	{
+		DefaultEntityMetaData refMetaData = new DefaultEntityMetaData("RefEntityTable");
+		refMetaData.addAttribute("refAttr").setDataType(MolgenisFieldTypes.STRING).setIdAttribute(true)
+				.setNillable(false);
+
+		DefaultEntityMetaData metaData = new DefaultEntityMetaData("XrefTable");
+		metaData.addAttribute("id").setDataType(MolgenisFieldTypes.INT).setIdAttribute(true).setNillable(false);
+		metaData.addAttribute("xrefAttr").setDataType(MolgenisFieldTypes.XREF).setRefEntity(refMetaData)
+				.setNillable(false);
+
+		Repository refRepo = metaDataRepositories.addEntityMeta(refMetaData);
+		Repository repo = metaDataRepositories.addEntityMeta(metaData);
+
+		List<Entity> refEntityTableEntities = new ArrayList<>();
+		Entity refEntityTableEntity = new MapEntity("refAttr");
+		refEntityTableEntity.set("refAttr", "xref_value_1");
+		refEntityTableEntities.addAll(newArrayList(refEntityTableEntity));
+
+		List<Entity> xrefTableEntities = new ArrayList<>();
+		Entity xrefTableEntity = new MapEntity("xrefAttr");
+		xrefTableEntity.set("id", 1);
+		xrefTableEntity.set("xrefAttr", "xref_value_1");
+		xrefTableEntities.addAll(newArrayList(xrefTableEntity));
+
+		repo.deleteAll(); // cleanup
+		refRepo.deleteAll(); // cleanup
+	}
+
+	@Test
```

Comment:
Could probably also use a test that tests that the actual error message still appears for self-referencing non-nillable xref.